File: 0739-daily-temperatures/0739-daily-temperatures.cpp
class Solution {
public:
    vector<int> dailyTemperatures(vector<int>& temperatures) {
     
     int size = temperatures.size();
     stack<int> st;
     
     vector<int> answer(size, 0);
        
      for(int i = 0; i < size; i++)
      {          
         while(!st.empty() and temperatures[st.top()] < temperatures[i])
         {
             answer[st.top()] = i - st.top();
             st.pop();
         }
         st.push(i);
      }
      return answer;
    }
};

File: 0739-daily-temperatures/0739-daily-temperatures.py
class Solution:
    def dailyTemperatures(self, temperatures: List[int]) -> List[int]:
        
        
        stack = []
        answer = [0]*len(temperatures)
        for i in range(0, len(temperatures)):
            
            while(len(stack) > 0 and temperatures[stack[-1]] < temperatures[i]):
                answer[stack[-1]] = i - stack[-1]
                stack.pop()
            stack.append(i)
        return answer